apps/ledger-live-mobile/src/screens/FirmwareUpdate/index.tsx

```diff
@@ -509,6 +538,17 @@ export const FirmwareUpdate = ({
       );
     }
 
+    if (batteryStatusesState.error !== null) {
```

Comment:
Just: in that cas are we not displaying an error message if we get a state with `error: { type: "UnknownApdu" }` ? I don't see where we stop doing the battery check if the apdu is unknown ?

Reply:
If there is an `UnknownApdu` event type, there is no errors since it's an event with no other attribute than `type`

Reply:
But the `getBatteryStatusesAction` has an `error` prop in it's `state` (in `libs/ledger-live-common/src/deviceSDK/actions/getBatteryStatuses.ts`)

```TS
export type GetBatteryStatusesActionState = FullActionState<{
  batteryStatuses: (number | BatteryStatusFlags)[];
  error: { type: GetBatteryStatusesActionErrorType; message?: string; name?: string } | null;
}>;
```

And we update this state when receiving an event of type `taskError`:
```TS
case "taskError":
          return { ...initialState, error: { type: event.error } };
```

Which is the event (`taskError`) that we get from the command:
```TS
        catchError((err: Error) => {
          if (err instanceof TransportStatusError && err.statusCode === StatusCodes.UNKNOWN_APDU)
            return of<GetBatteryStatusesTaskEvent>({ type: "taskError", error: "UnknownApdu" });

          return throwError(() => err);
        }),
```

So at the end, we have a `batteryStatusesState` in `libs/ledger-live-common/src/deviceSDK/hooks/useBatteryStatuses.ts` that will be set with an error of type `UnknownApdu`

No ?

Reply:
If we have a `UnknownApdu` error, from `libs/ledger-live-common/src/deviceSDK/hooks/useBatteryStatuses.ts` we have an `event` and not an `error` in `getBatteryStatusesAction` `next` callback. So since we don't have any flags, we exit without setting `isBatteryLow`. And since it's an event, we don't have error either.

```
const sub = getBatteryStatusesAction({
        deviceId,
        statuses,
      }).subscribe({
        next: state => {
          setBatteryStatusesState(state);

          // no battery status flags available
          if (state.batteryStatuses.length <= 1) return;

          const [percentage, statusFlags] = state.batteryStatuses as [number, BatteryStatusFlags];

          if (percentage < lowBatteryPercentage && statusFlags.charging === 0) {
            setIsBatteryLow(true);
          }
        },
        complete: () => {
          setRequestCompleted(true);
        },
        error: err => log("error", "error while retrieving the battery statuses", err),
      });
```

So from `apps/ledger-live-mobile/src/screens/FirmwareUpdate/index.tsx`, `isBatteryLow=false` so no warning battery and since the error is handled as an event, no `batteryStatusesState.error` 😃